tests/foreman/api/test_bookmarks.py

```diff
@@ -0,0 +1,290 @@
+
+        @Assert: The new bookmark name is listed
+
+        @Status: Manual
+        """
+        for entity in BOOKMARK_ENTITIES:
+            with self.subTest(entity['controller']):
+                pass
+
+    @stubbed()
+    @tier1
+    def test_negative_update_bookmark_name(self):
+        """Update a bookmark with name already taken
+
+        @Feature: Scoped Search Bookmark Update
+
+        @Setup:
+
+        1. Create 2 bookmarks of random names with random query
+
+        @Steps:
+
+        1. Update the name to one of the other pre-created bookmarks
+
+        @Assert: Error - name already taken, bookmark not updated
+
+        @Status: Manual
+        """
+        for entity in BOOKMARK_ENTITIES:
+            with self.subTest(entity['controller']):
+                pass
+
+    @stubbed()
+    @tier1
+    def test_negative_update_bookmark_name_empty(self):
+        """Update a bookmark with an empty name
+
+        @Feature: Scoped Search Bookmark Update
+
+        @Setup:
+
+        1. Create a bookmark of a random name with random query
+
+        @Steps:
+
+        1. Update the name of the pre-created bookmark to an empty value
+
+        @Assert: Error - name cannot be empty, bookmark not updated
+
+        @Status: Manual
+        """
+        for entity in BOOKMARK_ENTITIES:
+            with self.subTest(entity['controller']):
+                pass
+
+    @stubbed()
+    @tier1
+    def test_positive_update_bookmark_query(self):
+        """Update a bookmark query
+
+        @Feature: Scoped Search Bookmark Update
+
+        @Setup:
+
+        1. Create a bookmark of a random name with random query
+
+        @Steps:
+
+        1. Update the query of the pre-created bookmark
+
+        @Assert: The updated query submitted
+
+        @Status: Manual
+        """
+        for entity in BOOKMARK_ENTITIES:
+            with self.subTest(entity['controller']):
+                pass
+
+    @stubbed()
+    @tier1
+    def test_negative_update_bookmark_query_empty(self):
+        """Update a bookmark with an empty query
+
+        @Feature: Scoped Search Bookmark Update
+
+        @Setup:
+
+        1. Create a bookmark of a random name with random query
+
+        @Steps:
+
+        1. Update the query of the pre-created bookmark
+
+        @Assert: Error - search query cannot be empty, bookmark not updated
+
+        @Status: Manual
+        """
+        for entity in BOOKMARK_ENTITIES:
+            with self.subTest(entity['controller']):
+                pass
+
+    @stubbed()
+    @tier1
+    def test_positive_update_bookmark_public(self):
+        """Update a bookmark public state
+
+        @Feature: Scoped Search Bookmark Update
+
+        @Setup:
+
+        1. Create 2 bookmarks of a random names with random queries, one public
+        and one private
+        2. Create a non-admin user with 'viewer' role
+
+        @Steps:
+
+        1. List the bookmarks using the pre-created user's credentials
```

Comment:
Hmmm just to see if I understood this:

1. Using the `admin` user, list the Bookmarks previously created. Assert that their `public` fields match the values used during their creation.
1. Update the Bookmarks previously created, setting their `public` values to the opposite value of their original values (ie. `True > False` and `False > True`)
1. Using the `non-admin` user, list the Bookmarks. Assert that their `public` fields match the new values used during the `update` steps